### PR TITLE
Replace phpro/zf-doctrine-hydration-module with doctrine/doctrine-laminas-hydrator

### PR DESCRIPTION
**Pull request by javabudd**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | yes
| QA            | no

### CODE REVIEW

**Comment by javabudd** [comment]
This PR branches off https://github.com/laminas-api-tools/api-tools-doctrine/pull/33 to attempt to fix CI builds for PHP 8

File: src/Server/Event/Listener/CollectionListener.php [hunk 1]
```diff
@@ -37,6 +36,8 @@
  */
 class CollectionListener implements ListenerAggregateInterface
 {
+    public const CONFIG_NAMESPACE = 'doctrine-hydrator';
```
Comment (javabudd) [inline src/Server/Event/Listener/CollectionListener.php:39]: This probably isn't the best place for this, thoughts?

File: src/Server/Resource/DoctrineResource.php [hunk 2]
```diff
@@ -294,24 +294,19 @@ public function getMultiKeyDelimiter()
...
     /**
      * @return HydratorInterface
      */
-    public function getHydrator()
+    public function getHydrator(): ?HydratorInterface
     {
         if (! $this->hydrator) {
             // FIXME: find a way to test this line from a created API.  Shouldn't all created API's have a hydrator?
-            $this->hydrator = new Hydrator\DoctrineObject($this->getObjectManager(), $this->getEntityClass());
+            $this->hydrator = new DoctrineObject($this->getObjectManager());
```
Comment (javabudd) [inline src/Server/Resource/DoctrineResource.php:309]: The 2nd argument passed into this constructor has always been incorrect. It was supposed to be whether to use byValue hydration or not.

**Comment by javabudd** [comment]
Declining in favor of https://github.com/laminas-api-tools/api-tools-doctrine/pull/29

**Comment by javabudd** [comment]
Reopening this PR as #29 has become stale and other PR's to address this issue are using incorrect hydrator libraries.

**Comment by ajgale** [comment]
+1 on this as it is still preventing much needed PHP 8 upgrade

**Comment by speby** [comment]
Also +1 on this

**Comment by datasage** [comment]
Please push this through. Will help get the rest of the project up to PHP 8.1/8.2

**Comment by davidwindell** [comment]
Is this likely to get merged anytime soon @Ocramius @weierophinney? We are starting a new API Tools project and would be good to have some direction. Thank you!

**Comment by davidwindell** [comment]
@javabudd I see you are still targeting PHP 8.0 (not 8.1). I guess this repo is abandoned now, so we'll roll our own Doctrine integration.

**Comment by TomHAnderson** [comment]
I was the original author of this repository.  And I'm a principal on doctrine-laminas-hydrator.  Since I left this project I've seen very little movement on this repository.  That is unfortunate.

I've moved on to GraphQL now.  @davidwindell if you're starting a new project I would encourage you to consider GraphQL.  My integration library is here:  https://github.com/API-Skeletons/doctrine-orm-graphql

**Comment by davidwindell** [comment]
Thanks for the ping @TomHAnderson. REST makes more sense for our use case but will certainly bear doctrine-orm-graphql in mind for future projects :+1:

**Comment by javabudd** [comment]
> I was the original author of this repository. And I'm a principal on doctrine-laminas-hydrator. Since I left this project I've seen very little movement on this repository. That is unfortunate.
> 
> I've moved on to GraphQL now. @davidwindell if you're starting a new project I would encourage you to consider GraphQL. My integration library is here: https://github.com/API-Skeletons/doctrine-orm-graphql

I would be more than happy to assist with this repository, hence the pull request, but it seems nobody is willing to review or merge PR's anymore.

**Comment by Ocramius** [comment]
FWIW: 

 * https://github.com/laminas/technical-steering-committee/blob/7bb1d85e560a4f247fae208da8ea8571e8d7c805/meetings/minutes/2023-01-09-TSC-Minutes.md?plain=1#L102-L158
 * https://github.com/laminas/technical-steering-committee/blob/7bb1d85e560a4f247fae208da8ea8571e8d7c805/meetings/minutes/2023-08-07-TSC-Minutes.md?plain=1#L62-L74
 * https://github.com/laminas/technical-steering-committee/blob/7bb1d85e560a4f247fae208da8ea8571e8d7c805/meetings/minutes/2023-10-02-TSC-Minutes.md?plain=1#L30-L53

We're spread very thin over here, heh.

**Comment by datasage** [comment]
I've been maintaining my own branches for forward compatibility. I can help with the effort needed to ensure that this repo is compatible with 8.2 and 8.3

**Comment by davidwindell** [comment]
> We're spread very thin over here, heh.

@Ocramius thank you for these meeting minutes. I am shocked to read the "laminas-mvc seems to be almost dead" comment. I guess I need to catch up with the times and start looking at Mezzio.

**Comment by davidwindell** [comment]
FYI we ended up going with API Platform.

**Pull request closed without merging**